src/Widgets/DisplayWidget.vala

```diff
@@ -87,6 +87,43 @@ public class Network.Widgets.DisplayWidget : Gtk.Box {
                 return true;
             });
             break;
+        case Network.State.CONNECTED_MOBILE_WEAK:
+            image.icon_name = "network-cellular-signal-weak-%ssymbolic".printf (secure? "secure-" : "");
+            break;
+        case Network.State.CONNECTED_MOBILE_OK:
+            image.icon_name = "network-cellular-signal-ok-%ssymbolic".printf (secure? "secure-" : "");
+            break;
+        case Network.State.CONNECTED_MOBILE_GOOD:
+            image.icon_name = "network-cellular-signal-good-%ssymbolic".printf (secure? "secure-" : "");
+            break;
+        case Network.State.CONNECTED_MOBILE_EXCELLENT:
+            image.icon_name = "network-cellular-signal-excellent-%ssymbolic".printf (secure? "secure-" : "");
+            break;
+        case Network.State.CONNECTING_MOBILE:
+            animation_timeout = Timeout.add (300, () => {
+                animation_state = (animation_state + 1) % 4;
```

Comment:
Can we use declare a different `uint` for this `animation_state` and also for the `animation_timeout`. This could very easily be broke because the same fields are used for WiFi image animation.